src/main/java/org/apache/sling/event/impl/jobs/JobImpl.java

```diff
@@ -312,14 +318,18 @@ public String update(final long eta) {
 
     public String log(final String message, final Object... args) {
         final String logEntry = MessageFormat.format(message, args);
-        final String[] entries = this.getProperty(Job.PROPERTY_JOB_PROGRESS_LOG, String[].class);
+        final ArrayDeque<String> entries = this.getProperty(Job.PROPERTY_JOB_PROGRESS_LOG, ArrayDeque.class);
```

Comment:
is this change backwards compatible? Is it possible to read properties of jobs, which have been written with a ```String[]``` value in here?

Reply:
Thanks, @joerghoh for pointing this out. I would make the necessary changes to fix this.